packages/medusa/src/repositories/product.ts

```diff
@@ -213,11 +213,13 @@ export const ProductRepository = dataSource.getRepository(Product).extend({
       customJoinBuilders: [
         (queryBuilder, alias, topLevel) => {
           if (topLevel === "variants") {
-            queryBuilder.leftJoinAndSelect(
-              `${alias}.${topLevel}`,
-              topLevel,
-              `${topLevel}.deleted_at IS NULL`
-            )
+            const joinMethod = select.filter(
+              (key) => !!key.match(/^variants\.\w+$/i)
```

Comment:
suggestion: can we use `include?('variants')` here?

Reply:
no we can't here, we don't want to take `variants.prices.amount` but just `variants.title` for example.